Foxtrot/Foxtrot/Utils/HelperMethods.cs

```diff
@@ -4186,8 +4186,7 @@ public static Method GetContractMethod(Method instantiatedMethod)
             
             if (candidate != null)
             {
-                if (candidate.Contract != null &&
-                    (candidate.Contract.RequiresCount > 0 || candidate.Contract.EnsuresCount > 0))
+                if (candidate.Contract != null && candidate.Contract.HasAssertions)
```

Comment:
Sounds like you fixed another bug (and one which might be much older) in addition to the original issue which deals only with async methods.